lib/withOnyx.js

```diff
@@ -67,12 +78,33 @@ export default function (mapOnyxToState) {
                 });
             }
 
+            /**
+             * This method is used externally by sendDataToConnection to prevent unnecessary renders while a component
+             * still in a loading state. The temporary initial state is saved to the component instance and setState()
+             * only called once all the necessary data has been collected.
+             *
+             * @param {String} statePropertyName
+             * @param {*} val
+             */
+            setInitialState(statePropertyName, val) {
+                this.tempState[statePropertyName] = val;
+
+                // All state keys should exist and at least have a value of null
+                if (!_.every(requiredKeysForInit, key => !_.isUndefined(this.tempState[key]))) {
+                    return;
+                }
+
+                this.setState({...this.tempState, loading: false}, () => {
+                    this.tempState = {};
+                });
```

Comment:
We don't need to reset `tempState` in a callback
This is equivalent and more performant

```js
this.setState({...this.tempState, loading: false});
this.tempState = {};
```

Also we could `delete this.tempState`